### PR TITLE
[nPMI] Fixed Run Sorting for Annotation and Height of Rows

### PR DESCRIPTION
**Pull request by Sparkier**
* Motivation for features / changes
Fixed incorrect soorting and patched annotations to correct height.

* Detailed steps to verify changes work correctly (as executed by you)
`bazel run tensorboard -- --logdir [your logdir]`
Go to http://localhost:6007/?experimentalPlugin=npmi

### CODE REVIEW

File: tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ export class AnnotationComponent implements AfterViewInit, OnChanges {
 
     this.yScale
       .rangeRound([0, this.chartHeight - this.runHeight])
-      .domain(this.runs.map((d: string) => d));
```
Comment (psybuzz) [inline tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts:179]: I'm curious why this changes.

Before, we are creating a copy of `this.runs` before passing it to D3's scale.domain().  Are we fixing cases where making copies was unnecessary?  If so, should we also update the `domain([...this.runs])` callsites below in this file?

File: tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts [hunk 2]
```diff
@@ -149,12 +149,12 @@ export class AnnotationComponent implements AfterViewInit, OnChanges {
...
-    this.runs = [...runs];
-    this.svg.style('height', `${this.runs.length * this.runHeight}px`);
-    this.height = this.runs.length * this.runHeight;
+    this.runs = this.activeRuns.filter((run) => runs.has(run));
+    this.svg.style('height', `${this.activeRuns.length * this.runHeight}px`);
     this.width = this.annotationContainer.nativeElement.clientWidth || 10;
     this.chartWidth = this.width - this.margin.left - this.margin.right;
-    this.chartHeight = this.height - this.margin.top - this.margin.bottom;
+    this.chartHeight =
+      this.runs.length * this.runHeight - this.margin.top - this.margin.bottom;
```
Comment (psybuzz) [inline tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts:157]: Is it intentional that `this.svg.style.height` is out of sync with `this.chartHeight`?  It's not clear to me why the former uses `activeRuns.length` and the latter uses `runs.length`.  Doesn't that lead to cases where the chartHeight is clipped?

File: tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts [hunk 3]
```diff
@@ -42,6 +42,7 @@ export class AnnotationComponent implements AfterViewInit, OnChanges {
   @Input() flaggedAnnotations!: string[];
   @Input() hiddenAnnotations!: string[];
   @Input() activeMetrics!: string[];
+  @Input() activeRuns!: string[];
```
Comment (psybuzz) [inline tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts:45]: Is it necessary for the AnnotationComponent to be aware of the difference between active and inactive runs?  If the component never renders both at the same time in the UI, perhaps we can reduce the component's concerns.

An alternative to adding this field, is to just filter the `@Input() data` being passed into the component to only include data for the active runs.

File: tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts (hunk 1 above)
Comment (Sparkier) [reply tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts:179]: The map function did not do anything and yes, there does not need to be a copy, that is why I removed it.

Agreed, the other copies also should not be made. Removed them.

File: tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts (hunk 2 above)
Comment (Sparkier) [reply tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts:157]: The SVG should always be `activeRuns.length * runHeight` to ensure all rows have the same height. I need to do this to be able to make use of `cdk-virtual-scroll-viewport`.

The drawing area which also determines the scales and thus y-positions of all d3 elements should, however, only consider the runs that the current annotation contains.

I added code comments for that.

File: tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts (hunk 3 above)
Comment (Sparkier) [reply tensorboard/webapp/plugins/npmi/views/annotations_list/annotation/annotation_component.ts:45]: It actually already does only include data for active runs.

I thus only need this for the aforementioned svg height calculation and changed this to be numActiveRuns.

**Review by psybuzz — APPROVED** [review]
